core-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/BatchWritableSink.java

```diff
@@ -47,18 +47,18 @@ protected BatchWritableSink(BatchReadableWritableConfig batchReadableWritableCon
 
   @Override
   public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
-    String datasetName = getProperties().get(Properties.BatchReadableWritable.NAME);
-    Preconditions.checkArgument(datasetName != null && !datasetName.isEmpty(), "Dataset name must be given.");
-    String datasetType = getProperties().get(Properties.BatchReadableWritable.TYPE);
-    Preconditions.checkArgument(datasetType != null && !datasetType.isEmpty(), "Dataset type must be given.");
-
-    Map<String, String> properties = Maps.newHashMap(getProperties());
-    properties.remove(Properties.BatchReadableWritable.NAME);
-    properties.remove(Properties.BatchReadableWritable.TYPE);
-
     // null check for tests, as macro fields map and properties are null in test cases
     if (batchReadableWritableConfig.getProperties() != null &&
       !batchReadableWritableConfig.containsMacro(Properties.BatchReadableWritable.NAME)) {
```

Comment:
Would just doing the ```!containsMacro(...)``` check be good enough? Name is not nullable so it would only be null in the case macro syntax was provided?